include/crocoddyl/core/constraints/residual.hxx

```diff
@@ -0,0 +1,301 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ConstraintModelResidualTpl<Scalar>::ConstraintModelResidualTpl(boost::shared_ptr<typename Base::StateAbstract> state,
+                                                               boost::shared_ptr<ResidualModelAbstract> residual,
+                                                               const VectorXs& lower, const VectorXs& upper)
+    : Base(state, residual,
+           2 * ((upper - lower).array() != 0.).count() -
```

Comment:
this is very inconvenient to read, and error prone if further modifications are needed. Please formulate it differently (ideally, initialize the ng and nh counters later).

Reply:
Done